### PR TITLE
fix toggle for public bounties

### PR DESCRIPTION
**Pull request by mattcasey**
I removed the call to retrieve pages when updating bounties. Conceptually, it's nice, but it takes too long to resolve the whole page tree. I think it is extremely unlikely (knock on wood) that someone would be moving bounty cards around while someone else is toggling this setting in a way that would break permissions.

### CODE REVIEW

File: lib/spaces/togglePublicBounties.ts [hunk 1]
```diff
@@ -4,22 +4,12 @@ import { v4, validate } from 'uuid';
...
 import { multiResolvePageTree } from 'lib/pages/server/resolvePageTree';
-import { hasSameOrMorePermissions, IPagePermissionWithSource } from 'lib/permissions/pages';
+import { hasSameOrMorePermissions } from 'lib/permissions/pages';
 
-import { DataNotFoundError, InvalidInputError } from '../utilities/errors';
+import { InvalidInputError } from '../utilities/errors';
 
 import type { PublicBountyToggle } from './interfaces';
 
-async function generatePublicBountyPermissionArgs({
```
Comment (mattcasey) [inline lib/spaces/togglePublicBounties.ts:13]: removed this so that we can call generatePublicBountyPermissionArgs() in one place, whether publicBountyBoard is true or false

File: lib/spaces/togglePublicBounties.ts [hunk 2]
```diff
@@ -150,41 +140,33 @@ export async function togglePublicBounties({ spaceId, publicBountyBoard }: Publi
...
-          publicBountyBoard
-        }
-      });
-
-      if (publicBountyBoard === true) {
-        const [deleteArgs, createArgs, childCreateArgs] = await generatePublicBountyPermissionArgs({
-          publicBountyBoard,
-          spaceId
-        });
+  const [deleteArgs, createArgs, childCreateArgs] = await generatePublicBountyPermissionArgs({
```
Comment (mattcasey) [inline lib/spaces/togglePublicBounties.ts:143]: moved this async code outside of the transaction

File: lib/spaces/interfaces.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import type { Space } from '@prisma/client';
 // If we use it, we will need to update the space-related hooks so they also use this subset. This way, we can guarantee consistent behaviour between logged-in and public mode.
 export type PublicSpaceInfo = Pick<Space, 'domain' | 'id'>;
 
-export interface PublicBountyToggle<T extends boolean = boolean> {
+export interface PublicBountyToggle {
```
Comment (motechFR) [inline lib/spaces/interfaces.ts:8]: Not sure why we had a T there in first place, but good cleanup!

File: lib/spaces/togglePublicBounties.ts (hunk 2 above)
Comment (motechFR) [reply lib/spaces/togglePublicBounties.ts:143]: Makes sense, good performance optimisation. A next level would be having much faster resolvePageTree in future :)

**Review by motechFR — APPROVED** [review]
@mattcasey Approved

Ran the unit tests, they're all green so that's good

We are getting a failing one, related to the error if space id doesn't exist. So we now get a PrismaError instead of a SystemError

Does seem like an edge case, so we could just delete this test. I think our API creates an unknown error from SystemError anyways

**Comment by mattcasey** [comment]
> Ran the unit tests, they're all green so that's good

We are getting a failing one, related to the error if space id doesn't exist. So we now get a PrismaError instead of a SystemError

Does seem like an edge case, so we could just delete this test. I think our API creates an unknown error from SystemError anyways

Thanks @motechFR ... i took a look and we can actually get a special code from prisma, and still return DatanotFound in that case
